site/content/docs/concepts/pipelines.en.md

```diff
@@ -35,31 +35,38 @@ Follow the three steps below, from your workspace root:
 
 ```bash
 $ copilot pipeline init
-$ git add copilot/pipeline.yml copilot/buildspec.yml copilot/.workspace && git commit -m "Adding pipeline artifacts" && git push
+$ git add copilot/ && git commit -m "Adding pipeline artifacts" && git push
 $ copilot pipeline deploy
 ```
+!!! Note
+    You need to have initiated at least one workload (service or job) before running `pipeline deploy`, so that your pipeline has something to deploy to your environments.
 
-✨ And you'll have a new pipeline configured in your application account. Want to understand a little bit more what's going on? Read on!
+✨ And you'll have a new pipeline configured __in your application account__. Want to understand a little bit more what's going on? Read on!
 
-## Setting up a Pipeline, step by step
+## Setting Up a Pipeline, Step By Step
 
-### Step 1: Configuring your Pipeline
+### Step 1: Configuring Your Pipeline
 
-Pipeline configurations are created at a workspace level. If your workspace has a single service, then your pipeline will be triggered only for that service. However, if you have multiple services in a workspace, then the pipeline will build all the services in the workspace. To start setting up a pipeline, `cd` into your service(s)'s workspace and run:
+Pipeline configurations are created at a workspace level. If your workspace has a single service, then your pipeline will be triggered only for that service. However, if you have multiple services in a workspace, then the pipeline will build all the services in the workspace. To start setting up a pipeline, `cd` into your app's workspace and run:
 
  `copilot pipeline init`
 
-This won't create your pipeline, but it will create some local files that will be used when creating your pipeline.
+This won't create your pipeline, but it will create some local files under `copilot/pipelines` that will be used when creating your pipeline.
 
-* __Release order__: You'll be prompted for environments you want to deploy to - select them based on the order you want them to be deployed in your pipeline (deployments happen one environment at a time). You may, for example, want to deploy to your `test` environment first, and then your `prod` environment.
+* __Pipeline name__: We suggest naming your pipeline `[repository name]-[branch name]` (press 'Enter' when asked, to accept the default name). This will distinguish it from your other pipelines, should you create multiple, and works well if you follow a pipeline-per-branch workflow.
+
+* __Release order__: You'll be prompted for environments you want to deploy to – select them based on the order you want them to be deployed in your pipeline (deployments happen one environment at a time). You may, for example, want to deploy to your `test` environment first, and then your `prod` environment.
+
+!!! Attention
```

Comment:
Maybe we should add a validation for this? Of course, envs can be added/deleted via the manifest after this step, but at least we could prevent some cases of it happening.
Opened an issue: #3424.

Reply:
> Do not create pipelines that follow the same branch and deploy to the same environment(s), as that will lead to a CloudFormation stack collision.

Why? from my understanding as long as they don't have the same pipeline name within the same app there won't be CFN stack naming collision 🤔

Reply:
It's not a name collision, it's a deployment collision with stack update....

Reply:
Didn't you have this issue when you were trying it out during the bug bash? WX was reporting your experience, then I had a similar one...?

Reply:
Oh yeah I see what you mean.

Reply:
I don't think Pipeline 2 will keep retrying; I believe it will fail and then the user has to manually 'Retry' via the console. We could make this a warning that that will happen rather than telling people not to do it if we think there is utility in being able to do this, even with the hiccup.

Reply:
I also tried the same thing and my observation was that an "UPDATE" by Pipeline 2 was triggered on `app-test-svc` immediately after the first update was finished, even though both pipelines are triggered at the same time 🤔

Reply:
Discussed offline. Removing warning for now.